docs/internals/optimizer.rst

```diff
@@ -458,6 +462,14 @@ is transformed to
 This transformation can also be useful when paired with LoopInvariantCodeMotion, since
 invariants in the loop-invariant conditions can then be taken outside the loop.
 
+Loops that already have a literal constant as iteration condition are not transformed.
+
+To avoid unnecessary rewriting, it is recommended to run this step after StructuralSimplifier.
+
+May destroy the :ref:`expression-split <expression-splitter>` form.
```

Comment:
I inferred the info about which steps preserve the expression-split and SSA forms from the current docs. Please double check that I did not make a mistake here. Since most steps seem to preserve the forms, I used the convention of stating it only if the step can destroy it.

For example this step (`ForLoopConditionIntoBody`) may destroy expression-split form because it would transform something like this:

```yul
for { ... } iszero(a) { ... } {
    ...
}
```

into this:
```yul
for { ... } 1 { ... } {
    if iszero(iszero(a)) { break }
}
```

`ExpressionSimplifier` also seems to mostly preserve it but also has some rules that don't. For example the `exp` rule will transform this:

```yul
let _0 := 0
let a := calldataload(_0)
let b := not(_0)
let c := exp(b, a)
sstore(a, c)
```

into this:
```yul
let _0 := 0
let a := calldataload(_0)
let b := 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
let c := sub(iszero(and(a, 1)), and(a, 1))
sstore(a, c)
```

Reply:
I would phrase it a bit differently, what do you think about

```suggestion
The :ref:`expression-split <expression-splitter>` form is not guaranteed to be preserved under this step.
```